src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -992,6 +1006,43 @@ pub extern "C" fn merge_directories(
   })
 }
 
+#[no_mangle]
+pub extern "C" fn run_local_interactive_process(
+  scheduler_ptr: *mut Scheduler,
+  request: Handle,
+) -> PyResult {
+  use std::process;
+
+  with_scheduler(scheduler_ptr, |scheduler| {
+    let types = &scheduler.core.types;
+    let construct_interactive_process_result = types.construct_interactive_process_result;
+
+    let argv: Vec<String> = externs::project_multi_strs(&request.into(), "argv");
+    if argv.is_empty() {
+      return Err("Empty argv list not permitted".to_string());
+    }
+
+    let mut command = process::Command::new(argv[0].clone());
+    for arg in argv[1..].iter() {
+      command.arg(arg);
+    }
+
+    // deliberately do-nothing SIGINT handler
+    ctrlc::set_handler(|| {}).map_err(|e| e.to_string())?;
```

Comment:
I think this will need to use [tokio-net's ctrl-c](https://github.com/tokio-rs/tokio/blob/master/tokio-net/src/signal/ctrl_c.rs): https://github.com/Detegr/rust-ctrlc/issues/30